SU2_CFD/src/output/filewriter/CSTLFileWriter.cpp

```diff
@@ -0,0 +1,332 @@
+      halo_nodes.push_back(dataSorter->GetElem_Connectivity(QUADRILATERAL, 0, i)-1);
+
+  /* Sorted list of halo nodes for this MPI rank. */
+  sorted_halo_nodes.assign(halo_nodes.begin(), halo_nodes.end());
+  sort(sorted_halo_nodes.begin(), sorted_halo_nodes.end());
+
+  /*--- We effectively tack the halo nodes onto the end of the node list for this partition.
+    TecIO will later replace them with references to nodes in neighboring partitions. */
+  num_halo_nodes = sorted_halo_nodes.size();
+  vector<long> neighbor_partitions(max<unsigned long>(1, num_halo_nodes));
+  vector<long>      neighbor_nodes(max<unsigned long>(1, num_halo_nodes));
+
+  for(unsigned long i = 0; i < num_halo_nodes; ++i) {
+    int owning_rank = dataSorter->FindProcessor(sorted_halo_nodes[i]);
+    unsigned long node_number = sorted_halo_nodes[i] - dataSorter->GetNodeBegin(owning_rank);
+    neighbor_partitions[i] = owning_rank;
+    neighbor_nodes[i] = static_cast<long>(node_number);
+  }
+
+  /* Gather halo node data. First, tell each rank how many nodes' worth of data we need from them. */
+  for (unsigned long i = 0; i < num_halo_nodes; ++i)
+    ++num_nodes_to_receive[neighbor_partitions[i]];
+  num_nodes_to_send.resize(size);
+  SU2_MPI::Alltoall(&num_nodes_to_receive[0], 1, MPI_INT, &num_nodes_to_send[0], 1, MPI_INT, MPI_COMM_WORLD);
+
+  /* Now send the global node numbers whose data we need,
+     and receive the same from all other ranks.
+     Each rank has globally consecutive node numbers,
+     so we can just parcel out sorted_halo_nodes for send. */
+  nodes_to_send_displacements.resize(size);
+  nodes_to_receive_displacements.resize(size);
+  nodes_to_send_displacements[0] = 0;
+  nodes_to_receive_displacements[0] = 0;
+
+  for(int iRank = 1; iRank < size; ++iRank) {
+    nodes_to_send_displacements[iRank]    = nodes_to_send_displacements[iRank - 1]    + num_nodes_to_send[iRank - 1];
+    nodes_to_receive_displacements[iRank] = nodes_to_receive_displacements[iRank - 1] + num_nodes_to_receive[iRank - 1];
+  }
+
+  int total_num_nodes_to_send = nodes_to_send_displacements[size - 1] + num_nodes_to_send[size - 1];
+
+  nodes_to_send.resize(max(1, total_num_nodes_to_send));
+  /* The terminology gets a bit confusing here. We're sending the node numbers
+     (sorted_halo_nodes) whose data we need to receive, and receiving
+     lists of nodes whose data we need to send. */
+  if (sorted_halo_nodes.empty()) sorted_halo_nodes.resize(1); /* Avoid crash. */
+  SU2_MPI::Alltoallv(&sorted_halo_nodes[0], &num_nodes_to_receive[0], &nodes_to_receive_displacements[0], MPI_UNSIGNED_LONG,
+                     &nodes_to_send[0],     &num_nodes_to_send[0],    &nodes_to_send_displacements[0],    MPI_UNSIGNED_LONG,
+                     MPI_COMM_WORLD);
+
+  /* Now actually send and receive the data */
+  data_to_send.resize(max(1, total_num_nodes_to_send * (int)fieldnames.size()));

```

Comment:
You missed a max<> here.

Reply:
I needed a `max<unsigned long>` or `1ul` in the other instances because the compiler threw an error because the  naked `1` seems to be interpreted as an <int>. Here this is `max<int>`, so the compiler does not need it, but I added that for consistency

Reply:
Because you have an old style cast to int in fieldnames.size(), what I was suggesting was to remove that cast in favor of promoting 1 to unsigned long (either with 1ul or the max<>) to be consistent with the other parts.

Reply:
True, it is now a max<unsigned long> as well and the old style cast is removed of course.